app/src/lib/stores/pull-request-store.ts

```diff
@@ -60,13 +135,68 @@ export class PullRequestStore {
     }
 
     const table = this.db.pullRequests
-    const insertablePRs = pullRequests.map(x => {
-      return { repoId, ...x }
-    })
+
+    const insertablePRs = new Array<IPullRequest>()
+    for (const pr of pullRequests) {
+      let headRepo: GitHubRepository | null = null
+      if (pr.head.repo) {
+        headRepo = await this.repositoriesStore.findOrPutGitHubRepository(
+          repository.endpoint,
+          pr.head.repo
+        )
+      }
+
+      // We know the base repo isn't null since that's where we got the PR from
+      // in the first place.
+      const baseRepo = await this.repositoriesStore.findOrPutGitHubRepository(
+        repository.endpoint,
+        pr.base.repo!
```

Comment:
Same thing here - can we avoid the `!` by getting TS to trust the validity of `pr.base.repo` by extracting to a local variable?